src/prelude/string.savi

```diff
@@ -346,29 +341,36 @@
 
   :fun val each_split(split_byte U8) // TODO: a byte shouldn't be the only thing we can split by...
     :yields String
-    finish USize = 0
-    start = finish
-    while (finish < @_size) (
-      byte = @_ptr._get_at(finish)
-      if (byte == split_byte) (
-        yield @val_from_cpointer(
-          @_ptr._offset(start)
-          (finish - start)
-          @_space - start
-        )
-        finish = finish + 1
-        start = finish
+    start USize = 0
+    while (start < @_size) (
+      try (
+        split_pos = @_index_of!(split_byte, start)
+        yield @_slice(start, split_pos)
+        start = split_pos + 1
       |
-        finish = finish + 1
+        yield @_slice(start, @_size)
+        break
       )
     )
-    yield @val_from_cpointer(
-      @_ptr._offset(start)
-      (finish - start)
-      @_space - start
-    )
     @
 
+  :fun val split2!(split_byte U8) StringPair
```

Comment:
I'm not wild about the `split2` name but I have nothing confident to offer to replace it at this time, so we can move forward with this for now.